pkg/link/opstate.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package link
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/vishvananda/netlink"
+)
+
+func WaitForOperStateUp(linkName string) (netlink.Link, error) {
+	var link netlink.Link
+	var err error
+	retries := []int{0, 50, 500, 1000, 1000}
+	for idx, sleep := range retries {
+		time.Sleep(time.Duration(sleep) * time.Millisecond)
+
+		link, err = netlink.LinkByName(linkName)
+		if err != nil {
+			return nil, err
```

Comment:
let's be consistent and return here fmt.Errorf("failed to retrieve link: %v", err) as in SetUp